source/core_http_client.c

```diff
@@ -1340,9 +1340,9 @@ static HTTPStatus_t addHeader( HTTPRequestHeaders_t * pRequestHeaders,
             pBufferCur += fieldLen;
 
             /* Copy the field separator, ": ", into the buffer. */
-            ( void ) strncpy( pBufferCur,
```

Comment:
What warning are we seeing here?  Can you add the warning in the description of the PR ? Thanks.

Reply:
HTTP_HEADER_FIELD_SEPARATOR_LEN == 2 and  HTTP_HEADER_FIELD_SEPARATOR is a string of length 2 + null terminator.

For proper usage, the third argument to strncpy here should be the length of pBufferCur. For our use case though (where it is known that the buffer is zero-filled), using strncpy is not necessary.

```
/project/aws-iot-device-sdk-embedded-C/libraries/standard/coreHTTP/source/core_http_client.c:1361:22: error: 'strncpy' output truncated before terminating nul copying 4 bytes from a string of the same length [-Werror=stringop-truncation]
 1361 |             ( void ) strncpy( pBufferCur,
      |                      ^~~~~~~~~~~~~~~~~~~~
 1362 |                               HTTP_HEADER_END_INDICATOR,
      |                               ~~~~~~~~~~~~~~~~~~~~~~~~~~
 1363 |                               HTTP_HEADER_END_INDICATOR_LEN );```